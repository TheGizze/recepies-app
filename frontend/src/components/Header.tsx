import React, { useState, useEffect } from 'react';
import * as styles from './styles/header.module.scss';
import { routes } from '../routes';
import { Link } from 'react-router-dom';

const NavigationLinks = () => {
  const [usedRoutes, setUsedRoutes] = useState<RouteType[]>([]);

  useEffect(() => {
    setUsedRoutes(
      routes
        .filter(route => route.linkLocation === 'Header')
        .sort((linkA, linkB) => {
          if (linkA.order > linkB.order) return 1;
          if (linkB.order > linkA.order) return -1;
          return 0;
        })
    );
  }, [routes]);

  if (usedRoutes.length > 0) {
    return (
      <>
        {usedRoutes.map(route => {
          return (
            <span className={styles.linkWrapper}>
              <Link key={route.path} to={route.path}>
                {route.linkText}
              </Link>
            </span>
          );
        })}
      </>
    );
  }
  return <span />;
};

const Header: React.FunctionComponent = () => {
  return (
    <div className={styles.headerWrapper}>
      <div className={styles.logoWrapper}>
        <Link className={styles.logo} to={'/'}>
          <h1>Our logo</h1>
        </Link>
      </div>
      <div className={styles.navigationLinks}>
        <NavigationLinks />
      </div>
    </div>
  );
};

export { Header };
