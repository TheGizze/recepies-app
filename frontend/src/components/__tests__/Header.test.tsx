import React from 'react';
import { render } from '@testing-library/react';
import { Header } from '../Header';
import { MemoryRouter } from 'react-router-dom';

const TestWrapper = () => {
  return (
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  );
};

describe('Header tests', () => {
  it('Should match snapshot', () => {
    const { container } = render(<TestWrapper />);
    expect(container).toMatchSnapshot();
  });
});
