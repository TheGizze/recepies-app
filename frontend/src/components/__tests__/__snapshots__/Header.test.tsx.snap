// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests Should match snapshot 1`] = `
<div>
  <div>
    <div>
      <a
        href="/"
      >
        <h1>
          Our logo
        </h1>
      </a>
    </div>
    <div>
      <span>
        <a
          href="/about"
        >
          About
        </a>
      </span>
    </div>
  </div>
</div>
`;
